File: Java/Problem_064.java
public class Euler64 {
    public static void main(String[] args) {
        final int LIMIT = 10000;
        int count = 0;

        for (int N = 2; N <= LIMIT; N++) {
            int a0 = (int)Math.sqrt(N);
            if ((long)a0 * a0 == N) continue;

            long m = 0, d = 1, a = a0;
            int period = 0;

            while (true) {
                m = d * a - m;
                d = (N - m * m) / d;
                a = (a0 + m) / d;
                period++;
                if (a == 2L * a0) break;
            }

            if ((period & 1) == 1) count++;
        }

        System.out.println(count); // 1322
    }
}

File: Python/Problem_064.py
import math

def euler_64(limit: int = 10_000) -> int:
    count = 0
    for N in range(2, limit + 1):
        a0 = int(math.isqrt(N))
        if a0 * a0 == N:
            continue

        m, d, a = 0, 1, a0
        period = 0

        while True:
            m = d * a - m
            d = (N - m * m) // d
            a = (a0 + m) // d
            period += 1
            if a == 2 * a0:
                break

        if period % 2 == 1:
            count += 1

    return count

if __name__ == "__main__":
    print(euler_64())  # 1322
